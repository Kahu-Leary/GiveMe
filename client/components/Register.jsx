import React, { useState, useContext } from 'react'
import { register, isAuthenticated } from 'authenticare/client'

// import { UserContext, updateUserContext } from './UserContext'

function Register (props) {
//  const [, setUser] = useContext(UserContext)
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')
  const [id, setUserId] = useState('')
  const [email, setUserEmail] = useState('')
  const baseUrl = '/api/v1'

  const handleClick = () => {
    register(
      {
        username: username,
        password: password,
        // email: email,
        id: Number(id)
        // email: email
      },
      { baseUrl }
    )
      .then(() => {
        if (isAuthenticated()) {
          // updateUserContext(setUser)
          return props.history.push('/giveMe')
        }
        return null
      })
      .catch((error) => {
        console.log('error: ', error.message)
      })
  }

  return (
    <div className="registerForm" >
      <div>
        <img className="is-rounded" src='./img/logo-alt.png' />
      </div>
      <div className="leftRegister">
        <div className="registerTitle">
          <h1>Register</h1>
        </div>
        <label className="label">email</label>
        <input className="input" type="email" placeholder="email" onChange={(e) => setUserEmail(e.target.value)}></input>

        <label className="label">Username</label>
        <input className="input" type="username" value={username} placeholder="username" onChange={(e) => setUsername(e.target.value)}></input>

        <label className="label">Password</label>
        <input className="input" type="password" value={password} placeholder="password" onChange={(e) => setPassword(e.target.value)}></input>

        <button type="button" className="button is-primary" onClick={handleClick} data-testid="submitButton">Register</button>
      </div>
    </div>
  )
}

export default Register
