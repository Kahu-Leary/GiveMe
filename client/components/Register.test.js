import React from 'react'
import { HashRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { UserProvider } from './UserContext'

import Register from './Register'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <HashRouter>
        <UserProvider>
          < Register/>
        </UserProvider>
      </HashRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
