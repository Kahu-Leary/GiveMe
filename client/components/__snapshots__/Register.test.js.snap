// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form"
>
  <div
    className="formLeft"
  >
    <img
      className="logoImg"
      src="./img/logo-txt-orange.png"
    />
  </div>
  <div
    className="formRight"
  >
    <h1>
      Register
    </h1>
    <input
      className="formInput"
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
    <input
      className="formInput"
      onChange={[Function]}
      placeholder="Username"
      type="username"
      value=""
    />
    <input
      className="formInput"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <button
      className="button primaryBtn"
      data-testid="submitButton"
      onClick={[Function]}
      type="button"
    >
      Register
    </button>
    <a
      className=""
      href="#/signin"
      onClick={[Function]}
    >
      Sign In
    </a>
  </div>
</div>
`;
